ext/standard/basic_functions.c

```diff
@@ -1713,27 +1713,17 @@ static int user_shutdown_function_call(zval *zv) /* {{{ */
 
 static void user_tick_function_call(user_tick_function_entry *tick_fe) /* {{{ */
 {
-	zval retval;
-	zval *function = &tick_fe->arguments[0];
-
-	/* Prevent reentrant calls to the same user ticks function */
-	if (! tick_fe->calling) {
-		tick_fe->calling = 1;
-
-		if (call_user_function(NULL, NULL,
-								function,
-								&retval,
-								tick_fe->arg_count - 1,
-								tick_fe->arguments + 1
-								) == SUCCESS) {
-			zval_ptr_dtor(&retval);
-		} else {
-			zend_string *function_name = zend_get_callable_name(function);
-			zend_throw_error(NULL, "Registered tick function %s() cannot be called, function does not exist", ZSTR_VAL(function_name));
-			zend_string_release(function_name);
-		}
+	/* Prevent re-entrant calls to the same user ticks function */
+	if (!tick_fe->calling) {
+		zval tmp;
 
-		tick_fe->calling = 0;
+		/* set tmp zval */
+		tick_fe->fci.retval = &tmp;
+
+		tick_fe->calling = true;
+		zend_call_function(&tick_fe->fci, &tick_fe->fci_cache);
+		zend_release_fcall_info_cache(&tick_fe->fci_cache);
```

Comment:
@nikic do I actually need to call `zend_release_fcall_info_cache()` here?

Reply:
I don't think so. zend_call_function should be releasing trampolines itself.

Reply:
What you do need to do though is dtor &tmp, otherwise the return value will leak.

Reply:
Hum, I didn't have a leak with ASAN so I forgot to do that